services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3720,14 +3720,19 @@ Pair<Store, Version> waitVersion(String clusterName, String storeName, int versi
   @Override
```

Comment:
@smaji comment in a random place to start a thread:

> While you are at it can you fix the code in `transmitVersionSwapMessage` as well ?
This function will not get triggered in parent controller anymore. Do we still need a fix there?

Reply:
It would be nice to block it in parent from accidental usage.

Reply:
I don't know about that. I'd rather not put band-aids to functions that weren't written to be executed in the parent controller. As I see it, entering `transmitVersionSwap` function in parent controller is an issue in itself. So, throwing an exception is the right thing to do

I think we need to refactor our `VeniceHelixAdmin` and `VeniceParentHelixAdmin` logics to be better structured with better protection. There are a lot of chances of unintentionally doing the wrong thing since `VeniceHelixAdmin` runs in both child and parent controllers. We generally forget to handle "Parent controller" logic in this `VeniceHelixAdmin`

Reply:
Yeah, I see `rollForwardToFutureVersion` and backup version are also unprotected in `VeniceHelixAdmin` from parent controller running it. throwing exception also block the admin channel, no ?

Reply:
Yeah. Throwing exception will block the channel. I think it is expected to have these two functions in `VeniceHelixAdmin` as currently, we support a single colo mode of deployment (most tests use this)